File: GridView/EFCodeFirstInHierarchy/RadGridViewEFCodeFirstInHierarchyVB/RadGridViewEFCodeFirst.Data/Contracts/IRadGridViewEFCodeFirstContext.vb
Imports System.Data.Entity
Imports RadGridViewEFCodeFirst.Models
Imports System.Data.Entity.Infrastructure

Public Interface IRadGridViewEFCodeFirstContext
    Property Orders() As IDbSet(Of Order)

    Property OrderTypes() As IDbSet(Of OrderType)

    Property Shippers() As IDbSet(Of Shipper)

    Function [Set](Of T As Class)() As IDbSet(Of T)

    Function Entry(Of T As Class)(entity As T) As DbEntityEntry(Of T)

    Sub SaveChanges()
End Interface

File: GridView/EFCodeFirstInHierarchy/RadGridViewEFCodeFirstInHierarchy/RadGridViewEFCodeFirst.Data/Contracts/IRadGridViewEFCodeFirstContext.cs
using System;
using System.Data.Entity;
using System.Data.Entity.Infrastructure;
using System.Linq;

using RadGridViewEFCodeFirst.Models;

namespace RadGridViewEFCodeFirst.Data.Contracts
{
    public interface IRadGridViewEFCodeFirstContext
    {
        IDbSet<Order> Orders { get; set; }

        IDbSet<OrderType> OrderTypes { get; set; }

        IDbSet<Shipper> Shippers { get; set; }

        IDbSet<T> Set<T>() where T : class;

        DbEntityEntry<T> Entry<T>(T entity) where T : class;

        void SaveChanges();
    }
}
